lua/octo/pickers/telescope/provider.lua

```diff
@@ -268,50 +268,71 @@ end
 
 function M.pull_requests(opts)
   opts = opts or {}
-  if not opts.states then
-    opts.states = "OPEN"
+  local cfg = octo_config.values
+
+  if not opts.limit then
+    opts.limit = cfg.pull_requests.limit
   end
-  local filter = get_filter(opts, "pull_request")
-  if utils.is_blank(opts.repo) then
-    opts.repo = utils.get_remote_name()
+
+  utils.info "Fetching pull requests (this may take a while) ..."
+  local args = {
+    "pr", "list",
+    "--limit", tostring(opts.limit),
+    "--json", "author,number,title,url,headRepository,headRepositoryOwner,headRefName,isDraft"
+  }
+  if opts.repo then
+    table.insert(args, "-R")
+    table.insert(args, opts.repo)
   end
-  if not opts.repo then
-    utils.error "Cannot find repo"
-    return
+  local keys = { "author", "assignee", "label", "state", "head", "base", "search" }
+
+  for _, key in ipairs(keys) do
+    if opts[key] then
+      table.insert(args, "--" .. key)
+      table.insert(args, opts[key])
+    end
   end
 
-  local owner, name = utils.split_repo(opts.repo)
-  local cfg = octo_config.values
-  local order_by = cfg.pull_requests.order_by
-  local query =
-    graphql("pull_requests_query", owner, name, filter, order_by.field, order_by.direction, { escape = false })
-  utils.info "Fetching pull requests (this may take a while) ..."
   gh.run {
-    args = { "api", "graphql", "--paginate", "--jq", ".", "-f", string.format("query=%s", query) },
+    args = args,
+
     cb = function(output, stderr)
       if stderr and not utils.is_blank(stderr) then
         utils.error(stderr)
       elseif output then
-        local resp = utils.aggregate_pages(output, "data.repository.pullRequests.nodes")
-        local pull_requests = resp.data.repository.pullRequests.nodes
+        local pull_requests = vim.fn.json_decode(output)
         if #pull_requests == 0 then
           utils.error(string.format("There are no matching pull requests in %s.", opts.repo))
           return
         end
         local max_number = -1
+        local username_col_len = 0
+        local branch_name_col_len = 0
+        local authors = {}
+        local author_count = 0
         for _, pull in ipairs(pull_requests) do
-          if #tostring(pull.number) > max_number then
-            max_number = #tostring(pull.number)
-          end
+          -- modify result to be consistent with GraphQL output.
+          pull["__typename"] = "pull_request"
+          pull.author.username = pull.author.login
+          pull.repository = { nameWithOwner = pull.headRepositoryOwner.login .. "/" .. pull.headRepository.name }
+
+          authors[pull.author.id] = (authors[pull.author.id] or 0) + 1
```

Comment:
Im testing `Octo pr list` for the `neovim/neovim` repo and getting the following error in this line:

```
Error executing vim.schedule lua callback: ...tester/octo.nvim/lua/octo/pickers/telescope/provider.lua:319: table index is nil
stack traceback:
        ...tester/octo.nvim/lua/octo/pickers/telescope/provider.lua:319: in function 'cb'
        .../src/github.com/pwntester/octo.nvim/lua/octo/gh/init.lua:161: in function ''
        vim/_editor.lua: in function <vim/_editor.lua:0>
```

Seems like `id` may be nil and needs to be handled